src/darker/tests/test_linting.py

```diff
@@ -567,3 +580,35 @@ def test_get_messages_from_linters_for_baseline(git_repo):
         MessageLocation(a_py, 3): [LinterMessage("python", "Third line")],
     }
     assert result == expect
+
+
+class AssertEmptyStderrPopen(Popen[str]):
+    """A Popen to use for the following test; asserts that its stderr is empty"""
+    def __init__(
+        self,
+        args: List[str],
+        stdout: int,
+        encoding: str,
+        cwd: Path,
+        env: Dict[str, str],
+    ):
+        super().__init__(
+            args, stdout=stdout, stderr=PIPE, encoding=encoding, cwd=cwd, env=env
+        )
+        assert self.stderr is not None
+        assert self.stderr.read() == ""
+
+
+def test_get_messages_from_linters_for_baseline_no_mypy_errors(git_repo, capsys):
+    """Test for `linting._get_messages_from_linters_for_baseline`"""
```

Comment:
nit: Should it be something like

```suggestion
    """Ensure mypy does not fail early when __init__.py is at the repo root"""
```

in our codebase we sometimes also mention the bug ID (#498) in a comment somewhere. E.g., 

```
# Regression test for #498.
```